src/System.Xml.XmlDocument/src/System/Xml/Dom/XmlLoader.cs

```diff
@@ -523,12 +508,9 @@ private XmlAttribute LoadAttributeNodeDirect()
                 defattr.SetSpecified(false);
                 return defattr;
             }
-            else
-            {
-                attr = new XmlAttribute(r.Prefix, r.LocalName, r.NamespaceURI, _doc);
-                LoadAttributeValue(attr, true);
-                return attr;
-            }
+            attr = new XmlAttribute(r.Prefix, r.LocalName, r.NamespaceURI, _doc);
```

Comment:
I agree with @mellinoe: removing the else block makes it more difficult to understand the flow of the code just by looking at its structure. You have to spot the return statement at the end of the if block before you realize that the two blocks of code are mutually exclusive.

Can this be added to the code guidelines?